### PR TITLE
[14.0][IMP][hr_expense_invoice] Add smart button in invoice to navigate to the expense.

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This:
image

cc @JoanSForgeFlow Can you review?

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@victoralmau I can see that you have been a key contributor to this module. Perhaps you will be interested to review this PR.

File: hr_expense_invoice/__manifest__.py [hunk 1]
```diff
@@ -4,12 +4,12 @@
 
 {
     "name": "Supplier invoices on HR expenses",
-    "version": "14.0.1.1.1",
+    "version": "14.0.1.2.0",
```
Comment (victoralmau) [inline hr_expense_invoice/__manifest__.py:7]: It's not necessary.

File: hr_expense_invoice/models/account_move.py [hunk 2]
```diff
@@ -26,3 +28,21 @@ def _check_expense_ids(self):
...
                 )
+
+    def _compute_expense_count(self):
+        for rec in self:
+            rec.expense_count = len(rec.expense_ids)
+
+    def action_view_expenses(self):
+        self.ensure_one()
+        action = self.env.ref("hr_expense.hr_expense_actions_my_all")
+        result = action.sudo().read()[0]
```
Comment (victoralmau) [inline hr_expense_invoice/models/account_move.py:39]:
```suggestion
        result = self.env["ir.actions.act_window"]._for_xml_id(
            "hr_expense.hr_expense_actions_my_all"
        )
```

File: hr_expense_invoice/models/account_move.py [hunk 3]
```diff
@@ -26,3 +28,21 @@ def _check_expense_ids(self):
...
+        res = self.env.ref("hr_expense.hr_expense_view_form", False)
+        expenses = self.expense_ids
+        result["context"] = {}
+        if len(expenses) != 1:
+            result["domain"] = "[('id', 'in', " + str(expenses.ids) + ")]"
+        elif len(expenses) == 1:
+            result["views"] = [(res and res.id or False, "form")]
+            result["res_id"] = expenses.ids[0]
```
Comment (victoralmau) [inline hr_expense_invoice/models/account_move.py:47]:
```suggestion
        result["context"] = {}
        if self.expense_count > 1:
            result["domain"] = "[('id', 'in', self.expense_ids.ids)]"
        elif self.expense_count == 1:
            result["views"] = [(res and res.id or False, "form")]
            result["res_id"] = self.expense_ids.id
```

File: hr_expense_invoice/models/account_move.py [hunk 4]
```diff
@@ -26,3 +28,21 @@ def _check_expense_ids(self):
                         "linked to this invoice."
                     )
                 )
+
+    def _compute_expense_count(self):
+        for rec in self:
+            rec.expense_count = len(rec.expense_ids)
```
Comment (victoralmau) [inline hr_expense_invoice/models/account_move.py:34]:
```suggestion
    @api.depends("expense_ids")
    def _compute_expense_count(self):
        info_data = self.env["hr.expense"].read_group(
            [("invoice_id", "in", self.ids)], ["invoice_id"], ["invoice_id"]
        )
        mapping = {
            data["invoice_id"][0]: data["invoice_id_count"] for data in info_data
        }
        for rec in self:
            rec.expense_count = mapping.get(rec.id, 0)
```

File: hr_expense_invoice/views/account_move_views.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <div name="button_box" position="inside">
+                <button
+                    type="object"
+                    name="action_view_expenses"
+                    class="oe_stat_button"
+                    icon="fa-book"
+                    attrs="{'invisible': [('expense_count', '=', 0)]}"
+                    groups="account.group_account_user"
```
Comment (victoralmau) [inline hr_expense_invoice/views/account_move_views.xml:14]:
```suggestion
                    groups="hr_expense.group_hr_expense_manager"
```
It probably makes more sense to display button if you have access to expenses.

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Pull request closed without merging**